AzureMonitorAgent/ama_tst/modules/helpers.py

```diff
@@ -0,0 +1,116 @@
+import os
+import platform
+import subprocess
+
+# backwards compatible input() function for Python 2 vs 3
+try:
+    input = raw_input
+except NameError:
+    pass
+
+
+def get_input(question, check_ans, no_fit):
+    answer = input(" {0}: ".format(question))
+    while (not check_ans(answer.lower())):
+        print("Unclear input. {0}".format(no_fit))
+        answer = input(" {0}: ".format(question))
+    return answer
+
+
+def find_vm_distro():
+    """
+    Finds the Linux Distribution this vm is running on.
+    """
+    vm_dist = vm_id = vm_ver =  None
+    parse_manually = False
+    try:
+        vm_dist, vm_ver, vm_id = platform.linux_distribution()
+    except AttributeError:
+        try:
+            vm_dist, vm_ver, vm_id = platform.dist()
+        except AttributeError:
+            # Falling back to /etc/os-release distribution parsing
+            pass
+
+    # Some python versions *IF BUILT LOCALLY* (ex 3.5) give string responses (ex. 'bullseye/sid') to platform.dist() function
+    # This causes exception in the method below. Thus adding a check to switch to manual parsing in this case
+    try:
+        temp_vm_ver = int(vm_ver.split('.')[0])
+    except:
+        parse_manually = True
+
+    if (not vm_dist and not vm_ver) or parse_manually: # SLES 15 and others
+        try:
+            with open('/etc/os-release', 'r') as fp:
+                for line in fp:
+                    if line.startswith('ID='):
+                        vm_dist = line.split('=')[1]
+                        vm_dist = vm_dist.split('-')[0]
+                        vm_dist = vm_dist.replace('\"', '').replace('\n', '')
+                    elif line.startswith('VERSION_ID='):
+                        vm_ver = line.split('=')[1]
+                        vm_ver = vm_ver.replace('\"', '').replace('\n', '')
+        except:  # indeterminate OS
+            return None, None
+    return vm_dist.lower(), vm_ver.lower()
+
+
+def find_package_manager():
+    """
+    Checks if the dist is debian based or centos based and assigns the package manager accordingly
+    """
+    pkg_manager = ""
+    dist, _ = find_vm_distro()
+
+    dpkg_set = set(["debian", "ubuntu"])
```

Comment:
I originally went with this because I think it's how we check the package manager in agent.py, but there have definitely been issues with distros needing to be updated manually (and having a second static list here will likely cause more issues as we add distros).

One concern though is that I think Debian systems have `/etc/*-release` as well - doing the `/etc/debian_version` check first should make this a moot point, but is there a better way of confirming this (i.e. some sort of `/etc/redhat_version` file or similar)?

Reply:
I found on Redhat VMs there's a file `/etc/redhat_version`, so I have checks for those two files and if neither exist, I added a fallback to just try and run the `command -v <pkg manager>` command you mentioned in a previous comment.